src/conditions/base/contract.ts

```diff
@@ -6,31 +7,51 @@ import { RpcCondition, rpcConditionRecord } from './rpc';
 
 export const STANDARD_CONTRACT_TYPES = ['ERC20', 'ERC721'];
 
-const functionAbiVariable = Joi.object({
-  internalType: Joi.string(), // TODO is this needed?
-  name: Joi.string().required(),
-  type: Joi.string().required(),
-});
-
 const functionAbiSchema = Joi.object({
   name: Joi.string().required(),
   type: Joi.string().valid('function').required(),
-  inputs: Joi.array().items(functionAbiVariable),
-  outputs: Joi.array().items(functionAbiVariable),
-  // TODO: Should we restrict this to 'view'?
-  stateMutability: Joi.string(),
+  inputs: Joi.array(),
+  outputs: Joi.array(),
+  stateMutability: Joi.string().valid('view', 'pure').required(),
 }).custom((functionAbi, helper) => {
+  // Is `functionABI` a valid function fragment?
+  let asInterface;
+  try {
+    asInterface = new ethers.utils.Interface([functionAbi]);
+  } catch (e: unknown) {
+    const { message } = e as Error;
+    return helper.message({
+      custom: message,
+    });
+  }
+
+  if (!asInterface.fragments) {
```

Comment:
Should we be using `.functions` instead of `.fragments`? Just reading from here - https://docs.ethers.org/v5/api/utils/abi/interface/#Interface--properties.

Same for various logic below.

Reply:
:heavy_check_mark:

Reply:
Yes, we use them to fetch the `FunctionFragments`